contracts/Portfolio.sol

```diff
@@ -165,13 +165,20 @@ abstract contract PortfolioVirtual is Objective {
         // Checks
         if (to == address(this)) revert InvalidTransfer();
 
-        uint256 balance = getBalance(msg.sender, token);
-        if (amount == type(uint256).max) amount = balance;
-        if (amount > balance) revert DrawBalance();
+        uint8 decimals = IERC20(token).decimals();
+        uint256 amountWad = amount.scaleToWad(decimals);
+        uint256 balanceWad = getBalance(msg.sender, token);
+        if (amountWad == type(uint256).max) amountWad = balanceWad;
```

Comment:
In order to make life simple for callers (i.e. so they don't have to work backwards from `amountWad` to know what arg to pass to signal "use entire balance"), I'd restructure the code as:

```
uint256 amountWad;
uint256 balanceWad = getBalance(msg.sender, token);
if (amount == type(uint256).max) {
    amountWad = balanceWad;
} else {
    amountWad = amount.scaleToWad(decimals);
}
```